### PR TITLE
Add druntime options for code coverage

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**

### CODE REVIEW

**Comment by leandro-lucarella-sociomantic** [comment]
Depends on https://github.com/dlang/druntime/pull/1916

**Comment by leandro-lucarella-sociomantic** [comment]
I have no idea why it is marked as failed, the log says it was OK.

**Comment by leandro-lucarella-sociomantic** [comment]
And dlang/druntime#1916 was merged, so this one should be merged too, probably.

**Comment by leandro-lucarella-sociomantic** [comment]
http://dtest.dlang.io/results/5d16def9a81b5bc6c3bae15b7b4e3595efcb44fc/!base/

**Comment by leandro-lucarella-sociomantic** [comment]
Ah, sorry, it says "test failed".
```
Test failed: object.Exception@ae/sys/d/manager.d(801): Command ["timeout", "1800", "make", "-f", "posix.mak", "DIFFABLE=1", "test"] failed with status 2
----------------
/usr/local/src/phobos/std/exception.d:420 pure @safe void std.exception.bailOut!(Exception).bailOut(immutable(char)[], ulong, const(char[])) [0x7ec406]
```
I guess this means it timed out?

**Comment by wilzbach** [comment]
The problem is at the end:

```
/dev/shm/dtest/work/repo/dlang.org/web/code_coverage.html:216:<!--UNDEFINED MACRO: "CONSOLEV"-->
/dev/shm/dtest/work/repo/dlang.org/web/code_coverage.html:328:<!--UNDEFINED MACRO: "CONSOLEV"-->
/dev/shm/dtest/work/repo/dlang.org/web/code_coverage.html:332:<!--UNDEFINED MACRO: "CONSOLEV"-->
/dev/shm/dtest/work/repo/dlang.org/web/code_coverage.html:336:<!--UNDEFINED MACRO: "CONSOLEV"-->
```

**Comment by wilzbach** [comment]
Replaced `CONSOLEV` with the proper `CONSOLE` macro and rebased your PR.

**Comment by wilzbach** [comment]
Did another improvement by using the `SWITCH` macro.

Before:

image

After:

image

(the parameters are also clickable, relative links)

File: code_coverage.dd [hunk 1]
```diff
@@ -219,6 +219,48 @@ $(P When the $(DDSUBLINK dmd,switch-cov, $(B -cov)) switch is thrown,
...
+    $(SWITCH $(SWNAME dstpath),
+        Set path to where listing files are to be written (must exist).
+    )
+)
+
+$(H4 Example:)
+
+$(CONSOLE
+sieve --DRT-covopt="merge:1"
+mkdir reports
```
Comment (wilzbach) [inline code_coverage.dd]: Nit: I would use `mkdir -p` here.

File: code_coverage.dd [hunk 2]
```diff
@@ -219,6 +219,48 @@ $(P When the $(DDSUBLINK dmd,switch-cov, $(B -cov)) switch is thrown,
...
+sieve --DRT-covopt="merge:1 dstpath:reports"
+)
+
+$(P Will merge the results of this run with the previous run in the $(D
+sieve.lst) file in the current directory. While:)
+
+$(CONSOLE
+mkdir reports
```
Comment (wilzbach) [inline code_coverage.dd]: Nit: I would use `mkdir -p` here.

**Comment by wilzbach** [comment]
> I will have a look into it

Found the issue & a workaround -> https://github.com/dlang/dlang.org/pull/1952

**Comment by wilzbach** [comment]
(rebased to upstream to include the fixed `circleci.sh` script)

**Pull request merged**